BDArmory/Weapons/BDExplosivePart.cs

```diff
@@ -163,8 +172,13 @@ public override void OnStart(StartState state)
                 part.OnJustAboutToBeDestroyed += DetonateIfPossible;
                 part.force_activate();
                 sourcevessel = vessel;
+                if (part == null) return;
                 var MF = VesselModuleRegistry.GetModule<MissileFire>(vessel, true);
-                if (MF != null) sourcevessel = MF.vessel; // grab the vessel the Weapon manager is on at start
+                if (MF != null)
+                {
+                    sourcevessel = MF.vessel;
+                    sourcevesselTeam = MF.Team; // grab the vessel the Weapon manager is on at start
```

Comment:
This still doesn't deal with teams being rearranged after the spawning.
Can we set the team from the setter of `MissileBase.Team`? That's where missiles/multiMissiles/MMGs have their team set upon being fired.

Reply:
> This still doesn't deal with teams being rearranged after the spawning.
Missiles period don't respond to teamswitches; whatever team they had at launch is the one they have when they detonate.
We can swap it over to a MissileBase-esque Team setter, but that's not going to change the end result.

Reply:
Yes, that's the point I was trying to make. The team in `BDExplosivePart` should be being set when the part containing it is launched (the same as for missiles).
Setting the team in `OnStart` means that it's set when the vessel spawns, before it's assigned to the correct team, while setting the team in `DetonateIfPossible` was too late as then only the detonating exposives would have had their correct team assigned.